packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.ts

```diff
@@ -1087,24 +1092,67 @@ export class PrismaSchemaParserService {
         (property) => property.type === FIELD_TYPE_NAME
       ) as Field[];
 
-      const theOtherSide = remoteModelFields.find(
+      // from the remote model fields, find the fields that are relations to the current model (it can be more than one field)
+      const remoteRelatedFields = remoteModelFields.filter(
         (fieldItem: Field) =>
           formatModelName(model.name) ===
           formatModelName(fieldItem.fieldType as string)
       );
 
-      if (!theOtherSide) {
+      if (!remoteRelatedFields.length) {
         throw new Error(
           `The other side of the relation not found for field ${field.name} on model ${model.name}`
         );
       }
 
-      if (
-        this.isLookupField(schema, theOtherSide) &&
-        this.isNotAnnotatedRelationField(schema, theOtherSide) &&
-        theOtherSide.array
-      ) {
-        return true;
+      if (remoteRelatedFields.length === 1) {
+        if (
+          this.isLookupField(schema, remoteRelatedFields[0]) &&
+          this.isNotAnnotatedRelationField(schema, remoteRelatedFields[0]) &&
+          remoteRelatedFields[0].array
+        ) {
+          return true;
+        }
```

Comment:
Try to keep returns close to the logic to avoid potential issues due to unrelated changes to a later line in the function 
```suggestion
        if (
          this.isLookupField(schema, remoteRelatedFields[0]) &&
          this.isNotAnnotatedRelationField(schema, remoteRelatedFields[0]) &&
          remoteRelatedFields[0].array
        ) {
          return true;
        }
        return false;
```